### DISCUSSION TITLE
Unable to Understand the Respective NFT Challenges

### DISCUSSION THREAD

**Discussion by avalsonigara15** [Cyfrin/foundry-full-course-cu#2589] [Q&A] [answered]
I am not able to understand what the challenges are and what is to be done. I read the #2018 issue , I understood where to locate the challenges but what exactly is to be done is still unclear.
I only found a bunch of functions but don't know what is to be done with those functions.
I am unsure with what to expect with those functions.
Please Help

**Comment by avalsonigara15** [comment]
Where to find the Instructions to solve the Challenges?

**Reply by EngrPips** [reply]
Please show me what you have on your screen and I will guide you step by step on how to solve the challenege.

**Reply by avalsonigara15** [reply]
![image](https://github.com/user-attachments/assets/bec13dab-748f-41a2-894d-115acf31b7c1)
After this, according to #2018 issue I have to get the address of the challenge contract, but after that I don't know what is to be done.
where to get the instructions as of what is to be done in the challenge.

**Reply by EngrPips** [reply]
Hello @avalsonigara15, when you get the challenge contract look it up on etherscan and show me whatever you get when you look it up, and then I will guide you.

**Reply by avalsonigara15** [reply]
I got this
![image](https://github.com/user-attachments/assets/af9b28b5-ff75-4fdf-a6c0-d24a00dc2f84)
And this
![image](https://github.com/user-attachments/assets/12433edb-4cee-43a9-be8f-16122c07d29f)

**Reply by EngrPips** [reply]
click on the `solveChallenge` box in the second image, what did you get?

**Reply by avalsonigara15** [reply]
![image](https://github.com/user-attachments/assets/494433fb-779a-49ec-bd14-3ba344e587af)
I got this. over here I don't know what is location or what those Input function does?

**Reply by EngrPips** [reply]
Go back to the `code` tab and read the contract code to understand what `location` and `newLocation` is supposed to mean. Does this make sense to you?

**Reply by avalsonigara15** [reply]
On Solving the problem for the solidity 101 section 2 , I am getting this unusual transaction fee.
![image](https://github.com/user-attachments/assets/5cef8365-7ffc-41ed-8a4e-627108c4b4fb)

**Reply by avalsonigara15** [reply]
Also It is not that clear to understand like what is to  be done in the challenges , as I saw the challenges ahead as well. It's a bit difficult to understand them.

**Reply by EngrPips** [reply]
Please send the address of the challenge contract you are trying to solve. Let me attempt it and see how it plays out.

**Reply by avalsonigara15** [reply]
This is the contract address : 0x4A62A8901e6113dEF0AEeEC77E531779cd40da73

**Reply by EngrPips** [reply]
Alright, Thanks

**Reply by farman1094** [reply]
@EngrPips  have you give a try to add a challenge it also showing me unusual fees 
![image](https://github.com/user-attachments/assets/08c782ac-8750-48e9-a3e8-cc64cd97b3d8)

**Reply by EngrPips** [reply]
Hello @farman1094, You need to check that you haven't solved the challenge already with that same address you are using now as that will make it so your transaction will always fail. and secondly, what is the value you provided for the `location` parameter?

**Reply by EngrPips** [reply, marked as answer]
> This is the contract address : 0x4A62A8901e6113dEF0AEeEC77E531779cd40da73

Hello @avalsonigara15, You are meant to call the `solveChallenge` function on the `write` tab of the contract page on etherscan but to call the `solveChallenge` function, you need to provide the index of the `s_booleanArray`, which is currently set to `true`. You might ask how you are supposed to know the index of the `s_booleanArray` that is set to true, so what you should do is check the last successful transaction call to the contract and decode the input of the transaction you will see the index of the `s_booleanArray` that the person set to true. If you have gotten that index, then you choose any index within the `length -1` of the array and set it as the new index in `s_booleanArray`, which will be set to true. 

As of this moment, of you call the `solveChallenge` with the details in the image below, the transaction should be successful 

<img width="1440" alt="Screenshot 2024-10-07 at 09 50 15" src="https://github.com/user-attachments/assets/0755c546-b739-4fa2-a2ca-1458fe54ee8e">

**Reply by avalsonigara15** [reply]
This worked the transaction went through but my question is how will I or anyone will be able to figure out what is to be done. 
It is difficult to find any instructions regarding the challenge. How can I proceed further @EngrPips ?

**Reply by EngrPips** [reply]
Honestly, If you are unable to figure out the challenge it is a sign that you haven't understood properly everything that you have been taught so far and you should maybe redo the section.

**Reply by callghost** [reply]
> Hello @farman1094, You need to check that you haven't solved the challenge already with that same address you are using now as that will make it so your transaction will always fail. and secondly, what is the value you provided for the `location` parameter?

please @EngrPips , What did you mean by "same address you are using now as that will make it so your transaction will always fail" cause i am facing it too.

**Reply by callghost** [reply]
Hello @EngrPips , Clarify if i'm wrong. I have to use diff wallet address to solve the challenge, right? I just did now and i stopped getting the "transaction might fail".

**Reply by EngrPips** [reply]
Yeah, That means you had already solved the challenge with that address you were using before.

**Reply by callghost** [reply]
Thanks @EngrPips ,
I'm at lesson 4(Is it for the section 3 course).
I am having trouble solving it. I am yet to start section 3 of the course(Just finished 1 and 2).

**Reply by EngrPips** [reply]
> Thanks @EngrPips , I'm at lesson 4(Is it for the section 3 course). I am having trouble solving it. I am yet to start section 3 of the course(Just finished 1 and 2).

Hello @callgh0st, I don't understand this

**Reply by callghost** [reply]
Hello @EngrPips ,
I solved lesson 1, 2 and 3. I tired solving lesson 4 but i couldn't. I was wondering if the lesson 4 can be solved if i go through section 3 of solidity smart contract development? I just finished section 1 and 2.
![image](https://github.com/user-attachments/assets/768e7f32-493e-4cf0-90ef-7d5aa2d3534d)

**Reply by EngrPips** [reply]
Go through the lesson and then retry.

**Comment by callghost** [comment]
Hello @EngrPips ,
I'm having an issue with the "write". Once i input my twitter username, it pops out my metamask default interface.
![image](https://github.com/user-attachments/assets/2116aada-a80f-4076-913a-1043de3655c8)

**Reply by callghost** [reply]
> Hello @EngrPips , I'm having an issue with the "write". Once i input my twitter username, it pops out my metamask default interface. ![image](https://private-user-images.githubusercontent.com/153031754/373874200-2116aada-a80f-4076-913a-1043de3655c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxMzUzMzIsIm5iZiI6MTcyODEzNTAzMiwicGF0aCI6Ii8xNTMwMzE3NTQvMzczODc0MjAwLTIxMTZhYWRhLWE4MGYtNDA3Ni05MTNhLTEwNDNkZTM2NTVjOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwNVQxMzMwMzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZTM5NjFiN2NkNmU4NTg0YTdkMzBiMmJjNWI3NTQ4MTkyODIyZTEwNjVkNmNhNjhmMzk2MjI3ZTU4NmQ5NjE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.zeWC7cTQLtUs53v2LGzYeVoNyPftaKYfUKsfpym1cOo)

it seems the issue is from metamask.

![image](https://github.com/user-attachments/assets/cbe4ba60-287c-4ec7-a8ea-c57954b038cf)

**Reply by EngrPips** [reply]
Hello @callgh0st, when you click on `write` it is supposed to pop out your metamask so as to have you sign the transaction you want to make.

**Reply by callghost** [reply]
Hello @EngrPips , I'm already signed in. It seems the issue is with my metamask.

I disabled the extension then enabled it and it works. I was able to solve the challenge. 
![image](https://github.com/user-attachments/assets/bdeca618-86ea-4c08-a8a3-9a84b9bd6900)

Thanks for your assistance.

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.

**Comment by callghost** [comment]
Alright 👍
Thanks. Let me take a break then check it.

**Reply by EngrPips** [reply]
Yeah, Please always take breaks. They are very important.

**Reply by callghost** [reply]
Hello @EngrPips ,
Hope you're doing well!
I'm having an issue with lesson4. I already got the price but once i try to write the priceguess and username,  i got this:
We were not able to estimate gas. There might be an error in the contract and this transaction may fail.

I'm using different wallet address. I deployed the PriceCv3 using injector provider(metamask), deployed, got the price(6294100000000). The price is tooooo high.
![image](https://github.com/user-attachments/assets/d0c64768-7038-4cbc-ba39-33f8f455b3e2)

**Reply by EngrPips** [reply]
Hello @avalsonigara15, Is this a challenge you are trying to solve? If so, please send me the address of the challenge contract.

**Reply by callghost** [reply]
Hello @EngrPips ,
Here is the challenge I'm trying to solve 
0x1b30DA2a868704483143A4D46865Ac9585629fd0
Thanks.

**Reply by callghost** [reply]
@EngrPips ,
I have solved it. Thanks.

**Reply by EngrPips** [reply]
Awesome, You are giga chad. Keep crushing it.
